### PR TITLE
Add automated schema migration runner

### LINKED ISSUE #2374: Automate schema upgrades
In our CD environment we upgrade our Athenz instance regularly to the latest version. From time to time there are sql schema updates (mentioned in release notes) that breaks the upgrade process, and we need to manually resolve the issues by applying the necessary changes. It would be great if the server could apply the schema changes during startup.

This could be done with e.g. [Flywaydb](https://github.com/flyway/flyway), but there are likely other alternatives.

### PR DESCRIPTION
**Pull request by gjoranv**
# Description

Adds opt-in schema migration support that runs pending `update-YYYYMMDD.sql` files on server startup, eliminating the need to manually track and apply schema changes between Athenz versions.

Closes #2374

## How it works

A new `SchemaMigrationRunner` in `server_common` handles migrations for both ZMS and ZTS:

1. Scans the configured directory for `update-YYYYMMDD.sql` files
2. Creates a `schema_version` tracking table if it doesn't exist
3. Applies any files not yet recorded, in date order
4. Handles safe MySQL errors (duplicate column, table exists, etc.) gracefully, so it works on existing deployments where some migrations were already applied manually

Migrations are serialized across server instances using MySQL advisory locks.

## Configuration

Set the system property pointing to the schema updates directory. Migrations only run when the property is set.

- ZMS: `athenz.zms.schema_migration_dir=/path/to/servers/zms/schema/updates`
- ZTS: `athenz.zts.schema_migration_dir=/path/to/servers/zts/schema/updates`

## Safe for existing deployments

The runner is idempotent. On first run against an existing database, it attempts each migration file and treats "already applied" errors (duplicate column 1060, duplicate key 1061, table exists 1050, multiple primary key 1068) as success, recording them in `schema_version`. Subsequent startups skip all recorded versions.

## Testing

- 38 unit tests covering parsing, error handling, locking, and factory integration
- Integration test using Testcontainers MySQL: loads the base `zms_server.sql` schema, runs all 58 ZMS migration files, verifies idempotency on second run

## Changes

- `SchemaMigrationRunner` (new) - core migration logic in `server_common`
- `JDBCObjectStoreFactory` / `AWSObjectStoreFactory` - call runner for ZMS when configured
- `JDBCCertRecordStoreFactory` / `AWSCertRecordStoreFactory` - call runner for ZTS when configured
- `JDBCConsts` - new property constant

# Contribution Checklist:
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have read the [contribution guidelines](https://github.com/AthenZ/athenz/blob/master/CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request.**

### CODE REVIEW

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            current.append(line);
+            if (trimmed.endsWith(";")) {
+                addStatement(statements, current.toString());
+                current.setLength(0);
+            }
+        }
+
+        addStatement(statements, current.toString());
+        return statements;
+    }
```
[the remark this answers is not in the document]
Comment (gjoranv) [reply libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java:256]: All 66 existing migration files (58 ZMS + 8 ZTS) were audited - none use trailing comments after semicolons, multiple statements per line, semicolons in strings, or `#` comments. The integration test runs the parser against the entire real corpus and passes.

A full SQL lexer would add significant complexity for edge cases that do not exist in the actual migration files. If a future migration introduces one of these patterns, the parser can be hardened at that point.

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    static final String SELECT_APPLIED_VERSIONS =
+            "SELECT `version` FROM `schema_version` WHERE `success` = 1";
+
+    static final String INSERT_SCHEMA_VERSION =
+            "INSERT INTO `schema_version` (`version`, `script`, `execution_time_ms`, `success`) " +
+            "VALUES (?, ?, ?, ?) " +
+            "ON DUPLICATE KEY UPDATE `script` = VALUES(`script`), " +
+            "`execution_time_ms` = VALUES(`execution_time_ms`), " +
+            "`success` = VALUES(`success`), " +
+            "`installed_on` = CURRENT_TIMESTAMP(3)";
```
[the remark this answers is not in the document]
Comment (gjoranv) [reply libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java:74]: ZMS and ZTS use separate databases (`zms_server` vs `zts_store`), so each gets its own `schema_version` table naturally. Even if they were configured to share a database, the migration files have non-overlapping content (ZMS files alter `zms_server.*` tables, ZTS files alter `zts_store.*`), so there would be no version collision.

Happy to make the table name configurable if the maintainers see a concrete use case for it.

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (havetisyan) [reply libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java:256]: @gjoranv agree, we don't need a full sql lexer, but can you please update the instructions for the schema changes zms/schema/schema_update.md and zts/schema/schema_update.md and include restrictions how the file is generated so we don't end up with trailing comments, etc.

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (havetisyan) [reply libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java:74]: agree, I don't see a use case for configurable name.

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gjoranv) [reply libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/db/SchemaMigrationRunner.java:256]: Done, added formatting rules to both schema_update.md files.

**Review by havetisyan — APPROVED** [review]
